### PR TITLE
chore(release): add workflow for pr and release creation

### LINKED ISSUE #5785: Automate release PR
### The problem

The `automerge` action is not getting triggered when `package.json`s and `changelog`s get updated after a release. We've been manually opening a PR with these changes from the release branch the past two releases (see [here](https://github.com/carbon-design-system/ibm-products/pull/5783) for example).

### The solution

We should automate this and have it included as part of the release action.

### Design link

_No response_

### Links to other materials

_No response_

### Owner/main maintainer(s)

Our team

### Second/backup maintainer(s)

_No response_

### Product/offering

_No response_

### Business priority

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by annawen1**
Closes #5785 

This PR adds a github action workflow to create a PR to merge changelog and version bumps from the release branch to `main` as well as creating the release tag and notes.

I also edited the `release-base.yml` workflow:
 - changed the `DRY_RUN` env var name to `PUBLISH` instead as I felt that was more intuitive
 - removed the `create-release` flag for the lerna commands as the new workflow will create the releases instead
 - run the telemetry config update step only on dry-run so we're not running it twice
 

#### What did you change?
- create-release-tag-and-pr.yml
- release-base.yml
- get-changelog.js --> this is the script that picks up the commits between tags and organizes them for the release notes

#### How did you test and verify your work?

- Tested on my fork

### CODE REVIEW

File: scripts/get-changelog.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+function generateLog() {
+  let log = '';
+
+  log += getChangelog('@carbon/ibm-products', './packages/ibm-products');
+  log += getChangelog(
+    '@carbon/ibm-products-styles',
+    './packages/ibm-products-styles'
+  );
+
+  console.log(log);
```
Comment (matthewgallo) [inline scripts/get-changelog.js:174]: Do we still want to keep this in? Or can we remove if you tested this on your fork?

File: scripts/get-changelog.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    const commitParse = commit.replace(delimiter, '');
+    if (commit.startsWith('feat')) {
+      features.push(commitParse);
+    }
+
+    if (commit.startsWith('fix')) {
+      fixes.push(commitParse);
+    }
+
+    if (
```
Comment (matthewgallo) [inline scripts/get-changelog.js:120]: Can we add the other allowed commitlint types into `chores`? https://github.com/carbon-design-system/ibm-products/blob/main/package.json#L194-L206

**Comment by matthewgallo** [comment]
Also just one more general question, is the "create tag and PR" workflow just meant to be run via workflow dispatch, after every minor release?

**Comment by annawen1** [comment]
> Also just one more general question, is the "create tag and PR" workflow just meant to be run via workflow dispatch, after every minor release?

Yep, that's what I'm thinking as we'll need to pass in the appropriate tags to the workflow

File: scripts/get-changelog.js (hunk 1 above)
Comment (annawen1) [reply scripts/get-changelog.js:174]: I like to keep it in, so in the case we want to run it manually from command line we can copy and paste the output from the console log

File: scripts/get-changelog.js (hunk 2 above)
Comment (annawen1) [reply scripts/get-changelog.js:120]: ohh yep!

**Review by matthewgallo — APPROVED** [review]
This is great! 💯

**Review by davidmenendez — APPROVED** [review]
woot

**Comment by matthewgallo** [comment]
@all-contributors please add @annawen1 for code, infra

**Pull request merged**